### PR TITLE
Add clippy job

### LINKED ISSUE #231: Cargo clippy and cargo fmt in CI
- [x] use [this config](https://github.com/RoaringBitmap/roaring-rs/blob/master/rustfmt.toml)
- [ ] add `cargo clippy` and `cargo fmt` to [the CI](https://github.com/meilisearch/milli/blob/main/.github/workflows/test.yml)
- [ ] Fix clippy error in tests

### PR DESCRIPTION
**Pull request by unvalley**
# Pull Request

## Related issue
Fixes #231 

## What does this PR do?
- fix some clippy errors remain
- add clippy job to CI (I set `nightly` as toolchain)

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

### CODE REVIEW

**Comment by curquiza** [comment]
I put `hacktoberfest-accepted` to this one too, since it will be definitely helpful for this project once #659 is merged

File: milli/src/update/words_prefixes_fst.rs [hunk 1]
```diff
@@ -36,7 +36,7 @@ impl<'t, 'u, 'i> WordsPrefixesFst<'t, 'u, 'i> {
     /// Default value is `4` bytes. This value must be between 1 and 25 will be clamped
     /// to these bounds, otherwise.
     pub fn max_prefix_length(&mut self, value: usize) -> &mut Self {
-        self.max_prefix_length = value.min(25).max(1); // clamp [1, 25]
```
Comment (unvalley) [inline milli/src/update/words_prefixes_fst.rs:39]: Might be a reason that didn't use `clamp`.

File: milli/src/search/matches/matching_words.rs [hunk 2]
```diff
@@ -225,7 +225,7 @@ fn bytes_to_highlight(source: &str, target: &str) -> usize {
         for (col, char_t) in target.chars().enumerate() {
             let col = col + 1;
             let last_match_row = *last_row.get(&char_t).unwrap_or(&0);
-            let cost = if char_s == char_t { 0 } else { 1 };
+            let cost = usize::from(char_s != char_t);
```
Comment (unvalley) [inline milli/src/search/matches/matching_words.rs:228]: Converts a bool to a usize. The resulting value is 0 for false and 1 for true values.

[Examples](https://doc.rust-lang.org/std/primitive.usize.html#examples-87)
```rs
assert_eq!(usize::from(true), 1);
assert_eq!(usize::from(false), 0);
```

**Comment by unvalley** [comment]
@curquiza This pr is review ready. (Clippy is now success on CI)

**Review by curquiza — APPROVED** [review]
I approve the CI part.
@ManyTheFish is required for the code base part 😇

**Review by ManyTheFish — APPROVED** [review]
Hey @unvalley,
thank you for your contribution!

bors merge

**Pull request merged**